explore/src/main/scala/explore/targeteditor/SearchForm.scala

```diff
@@ -87,29 +109,37 @@ object SearchForm:
               Icons.ArrowDownLeft,
               trigger = Button(
                 severity = Button.Severity.Success,
-                disabled = props.readonly || props.searching.get.nonEmpty,
+                disabled = props.cloningTarget || props.searching.get.nonEmpty,
                 icon = Icons.Search,
                 loading = props.searching.get.nonEmpty,
-                onClick = error.setState(none) >> props.searching.mod(_ + props.id),
+                onClick = onButtonClick,
                 modifiers = List(^.untypedRef := buttonRef)
               ).tiny.compact,
               onSelected = targetWithId =>
                 (targetWithId.target match
                   case t @ Target.Sidereal(_, _, _, _) => props.targetSet(t)
                   case _                               => Callback.empty
                 ) >> searchComplete,
-              onCancel = searchComplete,
+              onCancel = updateNameIfNeeded >> searchComplete,
               initialSearch = term.get.some,
               showCreateEmpty = false
             )
           else Icons.Ban
 
-        val disabled = props.searching.get.exists(_ === props.id) || props.readonly
+        val disabled =
+          props.searching.get.exists(_ === props.id) || props.readonly || props.cloningTarget
 
-        // use a form here to handle submit? is it embedded?
         FormInputTextView(
           id = "search".refined,
-          value = term.withOnMod(props.targetName.set),
+          value = term.withOnMod(nes =>
+            // We submit to the singleEffect with a delay. If the user hit Enter or they
+            // click on the button before leaving the input field, this will get cancelled
+            // so that the name doesn't get updated and, more importantly, no target cloning
+            // occurs, which messed everything up.
+            singleEffect
+              .submit(IO.sleep(200.milliseconds) >> props.targetName.set(nes).toAsync)
```

Comment:
The Enter key scenario worked with much smaller sleeps, but it needed to be increased to handle the button clicking. The UI still feels fine.